pkg/engine/loadbalancers.go

```diff
@@ -132,6 +132,9 @@ func CreateMasterLoadBalancer(prop *api.Properties, isVMSS bool) LoadBalancerARM
 				},
 			},
 		}
+		if to.Bool(prop.OrchestratorProfile.KubernetesConfig.CloudProviderDisableOutboundSNAT) {
+			loadBalancingRules.LoadBalancingRulePropertiesFormat.DisableOutboundSnat = to.Bool(true)
```

Comment:
You also need to explicitly add outbound rules for TCP and UDP otherwise the cluster won't have outbound which is problematic (see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/536/files#diff-b3b16dd828cf19adbdc5be2d4e28d2d5R139)